File: proxy-pattern/java/src/RealImage.java
package src;

// 真实类实现基础接口
public class RealImage implements Image {

  private String fileName;

  public RealImage(String fileName) {
    // 在初始化时执行内部逻辑
    this.fileName = fileName;
    loadFromDisk(fileName);
  }

  @Override
  public void display() {
    System.out.println("RealImage::display() " + fileName);
  }

  // 这个方法只是内部使用
  private void loadFromDisk(String fileName) {
    System.out.println("RealImage::loadFromDisk()  " + fileName);
  }
}

File: proxy-pattern/python/src/RealImage.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

from src.Image import Image
from src.Image import Image


# 真实类实现基础接口
class RealImage(Image):

    real_image = None
    file_name = None

    def __init__(self, file_name):
        Image.__init__(self)
        self.file_name = file_name
        self.load_from_disk(file_name)

    def display(self):
        print('RealImage::display() ' + self.file_name)

    # 这个方法只是内部使用
    def load_from_disk(self, file_name):
        print('RealImage::load_from_disk()  ' + file_name)
